modules/security/keyvault/main.bicep

```diff
@@ -0,0 +1,98 @@
+@description('Deployment Location')
+param location string
+
+@description('Prefix of Cosmos DB Resource Name')
+param prefix string = 'kv'
+
+@description('Name of the Key Vault')
+param name string = '${prefix}-${uniqueString(resourceGroup().id)}'
+
+@description('The tenant ID where the Key Vault is deployed')
+param tenantId string = subscription().tenantId
+
+@description('For an existing Managed Identity, the Subscription Id it is located in')
+param subscriptionId string = subscription().subscriptionId
+
+@description('For an existing Managed Identity, the Resource Group it is located in')
+param resourceGroupName string = resourceGroup().name
+
+@description('Subnet ID for the Key Vault')
+param subnetID string = ''
+
+@description('Enable VNet Service Endpoints for Key Vault')
+param enableVNet bool = false
+
+@description('List of RBAC policies to assign to the Key Vault')
+param rbacPolicies array = []
+
+@description('RBAC Role Assignments to apply to each RBAC policy')
+param roleAssignments array = [
+  '4633458b-17de-408a-b874-0445c86b69e6' // rbacSecretsReaderRole
+  'a4417e6f-fecd-4de8-b567-7b0420556985' // rbacCertificateOfficerRole
+]
+
+@allowed([ 'new', 'existing' ])
+@description('Whether to create a new Key Vault or use an existing one')
+param newOrExisting string = 'new'
+
+@description('List of secrets to create in the Key Vault [ { secretName: string, secretValue: string }]')
+param secrets array = []
```

Comment:
Isn't this going to expose secrets?

Reply:
The only way to keep the value secure would be having it as a separate module and require secure string to be provided. So users consuming could call it per secret.

The secure object I found from research (we had internal request to use this for similar reasons) will keep the key-value pair secure until something requires that the key or the value be converted to plaintext, at that that point the whole object is converted to plaintext.

Reply:
im rolling back the complex object here. Let me see how far I can get only taking in on secret at a time.